google_auth_oauthlib/tool/__main__.py

```diff
@@ -0,0 +1,118 @@
+# Copyright (C) 2017 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Command line tool for fetching credentials using 3LO OAuth2 flow.
+
+This tool is intended to bootstrap development in situation when the
+application cannot easily run the 3LO OAuth2 flow, i.e: in an embedded
+device with limited input / display capabilities.
+
+This is not intended for production where the main application should
+run the 3LO OAuth2 flow to get authorization from the users.
+"""
+
+import json
+import os
+import os.path
+
+import click
+
+import google_auth_oauthlib.flow
+
+
+APP_NAME = 'google-oauthlib-tool'
+DEFAULT_CREDENTIALS_FILENAME = 'credentials.json'
+
+
+def credentials_flow_interactive(client_secrets_path, scopes, headless):
+    """Initiate an interactive OAuth2InstalledApp flow.
+
+    - If an X server is running: Run a browser based flow.
+    - If not: Run a console based flow.
+
+    Args:
+      client_secrets_path(str): The path to the client secrets JSON file.
+      scopes(Sequence[str]): The list of scopes to request during the flow.
+      headless(bool): If True, run a console based flow otherwise run a web
+      server based flow.
+
+    Returns:
+      google.oauth2.credentials.Credentials: new OAuth2 credentials authorized
+        with the given scopes.
+
+    """
+    flow = google_auth_oauthlib.flow.InstalledAppFlow.from_client_secrets_file(
+        client_secrets_path,
+        scopes=scopes)
+    if not headless:
+        flow.run_local_server()
+    else:
+        flow.run_console()
+    return flow.credentials
+
+
+def credentials_to_dict(credentials):
```

Comment:
should we use the same format as https://github.com/GoogleCloudPlatform/google-auth-library-python/blob/0c09c734ba8ed24f2c9f3ea8a1a190f29aa679b6/google/auth/_cloud_sdk.py#L83 ?